dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguagePluginHandler.java

```diff
@@ -195,36 +195,42 @@ public List<String> listFiles(String pathToDirectory) {
-            DescriptorLanguage.FileType type;
-            if (entry.getValue().genericFileType() == GenericFileType.IMPORTED_DESCRIPTOR) {
-                type = minimalLanguageInterface.getDescriptorLanguage().getFileType();
-            } else if (entry.getValue().genericFileType() == GenericFileType.TEST_PARAMETER_FILE) {
-                type = minimalLanguageInterface.getDescriptorLanguage().getTestParamType();
-            } else {
-                // For some languages this may be incorrect
-                type = FileType.DOCKERFILE;
-            }
-            if (minimalLanguageInterface.getDescriptorLanguage().isServiceLanguage()) {
-                // TODO: this needs to be more sophisticated
-                type = DescriptorLanguage.FileType.DOCKSTORE_SERVICE_YML;
+                // The language plugins don't necessarily set the file type
+                // so query the plugin to find out what the imported file
+                // type should be, because this is needed in downstream code.
+                // We assume if imported files are not descriptors or not test files they are Dockerfiles,
+                // however this may not be true for some languages, and we may have to change this
+                DescriptorLanguage.FileType type;
+                if (entry.getValue().genericFileType() == GenericFileType.IMPORTED_DESCRIPTOR) {
+                    type = minimalLanguageInterface.getDescriptorLanguage().getFileType();
+                } else if (entry.getValue().genericFileType() == GenericFileType.TEST_PARAMETER_FILE) {
+                    type = minimalLanguageInterface.getDescriptorLanguage().getTestParamType();
+                } else {
+                    // For some languages this may be incorrect
+                    type = FileType.DOCKERFILE;
+                }
+                if (minimalLanguageInterface.getDescriptorLanguage().isServiceLanguage()) {
+                    // TODO: this needs to be more sophisticated
+                    type = DescriptorLanguage.FileType.DOCKSTORE_SERVICE_YML;
+                }
+                String path = entry.getKey();
+                String fileContent = entry.getValue().content();
+                SourceFile sourceFile = SourceFile.limitedBuilder().type(type).content(fileContent).paths(path).build();
+                sourceFile.getMetadata().setTypeVersion(entry.getValue().languageVersion());
+                results.put(entry.getKey(), sourceFile);
             }
-            String path = entry.getKey();
-            String fileContent = entry.getValue().content();
-            SourceFile sourceFile = SourceFile.limitedBuilder().type(type).content(fileContent).paths(path).build();
-            sourceFile.getMetadata().setTypeVersion(entry.getValue().languageVersion());
-            results.put(entry.getKey(), sourceFile);
+            return results;
+        } catch (NullPointerException e) {
```

Comment:
What's motivation for catching only `NullPointerException` versus other types of `RuntimeException`s?

Reply:
It's an interesting question. The immediate answer is this is the type of error that I had and it killed the request. Fixing it gave some partial results ignoring imports for just the primary. 

Looking into other RuntimeExceptions, it seems like some would correspond to errors with the system, some would be plugin errors.